core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -558,6 +571,11 @@ public void testCommitStats() {
         assertThat(stats2.getUserData(), hasKey(Translog.TRANSLOG_UUID_KEY));
         assertThat(stats2.getUserData().get(Translog.TRANSLOG_GENERATION_KEY), not(equalTo(stats1.getUserData().get(Translog.TRANSLOG_GENERATION_KEY))));
         assertThat(stats2.getUserData().get(Translog.TRANSLOG_UUID_KEY), equalTo(stats1.getUserData().get(Translog.TRANSLOG_UUID_KEY)));
+        assertThat(Long.parseLong(stats2.getUserData().get(LocalCheckpointService.LOCAL_CHECKPOINT_KEY)), equalTo(0L));
+        assertThat(stats2.getUserData(), hasKey(GlobalCheckpointService.GLOBAL_CHECKPOINT_KEY));
+        assertThat(
+                Long.parseLong(stats2.getUserData().get(GlobalCheckpointService.GLOBAL_CHECKPOINT_KEY)),
+                equalTo(SequenceNumbersService.UNASSIGNED_SEQ_NO));
```

Comment:
maybe tell the engine of a global checkpoint and see that it is persisted? now we don't test this as global checkpoint stays the same.